### PR TITLE
Drop current state of the task on disable instances

### LINKED ISSUE #1624: Task CurrentState remain active if participant is not enabled
### Describe the bug
In Task Pipeline, since the liveinstance list contain the instance that is live and enabled, if the participant is disabled, then the task current state will remain unchanged and the controller assigns the task to another instance. In this case, once the old instance is reenabled, then the task would have two current state. This would cause scheduling and quota issues

### To Reproduce
Have a task running and disable one instance. Then you would notice that the current state remains unchanged,

### Expected behavior
The controller needs to make sure that the task has been dropped from the disabled instance before assigning it to new instance,

### PR DESCRIPTION
**Pull request by alirezazamani**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:
Fixes #1624 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
In this PR, if an instance is not enabled, the controller drops the current state of the task on the disabled node first before assigning the task to a new instance. Otherwise, once the instance becomes enable controller will see two running tasks.

### Tests

- [x] The following tests are written for this issue:
TestTaskCurrentStateDrop.testDropCurrentStateDisableInstance

- [x] The following is the result of the "mvn test" command on the appropriate module:
Helix-core:
```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR] org.apache.helix.integration.rebalancer.CrushRebalancers.TestCrushAutoRebalanceNonRack.testLackEnoughInstances(org.apache.helix.integration.rebalancer.CrushRebalancers.TestCrushAutoRebalanceNonRack)
[ERROR]   Run 1: TestCrushAutoRebalanceNonRack.testLackEnoughInstances:281 » Helix Failed to dr...
[ERROR]   Run 2: TestCrushAutoRebalanceNonRack.testLackEnoughInstances:273 » Helix Cluster CLUS...
[INFO] 
[INFO] 
[ERROR] Tests run: 1258, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:30 h
[INFO] Finished at: 2021-02-01T11:11:56-08:00
[INFO] ------------------------------------------------------------------------
```
The failed test passed when I run it individually.
mvn test -Dtest="TestCrushAutoRebalanceNonRack"
```
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 28.689 s - in org.apache.helix.integration.rebalancer.CrushRebalancers.TestCrushAutoRebalanceNonRack
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  34.777 s
[INFO] Finished at: 2021-02-01T11:28:43-08:00
[INFO] ------------------------------------------------------------------------
```

Helix-rest:
```
[INFO] Tests run: 171, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 102.54 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 171, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:48 min
[INFO] Finished at: 2021-02-01T11:30:59-08:00
[INFO] ------------------------------------------------------------------------
```

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java [hunk 1]
```diff
@@ -811,13 +816,23 @@ private static void addCompletedTasks(Set<Integer> set, JobContext ctx, Iterable
           || state == TaskPartitionState.DROPPED) {
         filteredTasks.add(partitionNumber);
       }
-      // Allow tasks whose assigned instances are no longer live for rescheduling
+      // Allow tasks whose assigned instances are no longer live or enable for rescheduling
       if (isTaskNotInTerminalState(state)) {
         String assignedParticipant = jobContext.getAssignedParticipant(partitionNumber);
+        final String pName = pName(jobResource, partitionNumber);
```
Comment (dasahcc) [inline helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java:822]: NIT : Let's not have the variable name same as the function name.

**Review by dasahcc — APPROVED** [review]
Overall, LGTM.

File: helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java (hunk 1 above)
Comment (alirezazamani) [reply helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java]: Thanks for the review. Fixed.

**Comment by alirezazamani** [comment]
This PR is ready to be merged.
Final commit message:

Drop current state of the task on disabled instances

In this commit, if an instance is not enabled, the controller drops
the current state of the task on the disabled node before assigning 
the task to another instance. Otherwise, once the instance becomes
enabled again the controller will see two running tasks.

**Pull request merged**